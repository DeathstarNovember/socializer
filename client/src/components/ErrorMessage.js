import React from "react";
import { Link } from "react-router-dom";
import { Button, Card } from "react-bootstrap";

const ErrorMessage = ({ message }) => {
  const is404 = message === "GraphQL error: Not found";
  return (
    <div className="d-flex justify-content-center">
      <Card>
        <Card.Body>
          <Card.Title>{is404 ? "Not found" : "Error"}</Card.Title>
          <Card.Text>
            {is404
              ? "The page you were looking for doesn't seem to exist."
              : "Something went wrong, sorry about that."}
          </Card.Text>
          <Link to="/">
            <Button variant="primary">Go home</Button>
          </Link>
        </Card.Body>
      </Card>
    </div>
  );
};

export default ErrorMessage;
