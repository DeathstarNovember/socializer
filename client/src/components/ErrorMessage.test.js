import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ErrorMessage from "./ErrorMessage";

describe("ErrorMessage", () => {
  it("renders correctly for not found error", () => {
    const { container } = render(
      <MemoryRouter>
        <ErrorMessage message="GraphQL error: Not found" />
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly for other error", () => {
    const { container } = render(
      <MemoryRouter>
        <ErrorMessage message="GraphQL error: Something" />
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
