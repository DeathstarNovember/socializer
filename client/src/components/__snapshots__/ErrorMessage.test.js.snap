// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly for not found error 1`] = `
<div
  class="d-flex justify-content-center"
>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Not found
      </div>
      <p
        class="card-text"
      >
        The page you were looking for doesn't seem to exist.
      </p>
      <a
        href="/"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          Go home
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`ErrorMessage renders correctly for other error 1`] = `
<div
  class="d-flex justify-content-center"
>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Error
      </div>
      <p
        class="card-text"
      >
        Something went wrong, sorry about that.
      </p>
      <a
        href="/"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          Go home
        </button>
      </a>
    </div>
  </div>
</div>
`;
